### PR TITLE
Association

### PR DESCRIPTION
**Pull request by yosefHesham**
**Changes Made:**

- Created book class.
- Created rental class.
- Created classroom class.
- Implemented relationships between classes.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr :label
```
Comment (bobb-Rob) [inline classroom.rb]: - [x] Great work on this project. According to the project requirement, you should create _**setter**_ and **_getter_** for `@label`.  This `attr :label` currently sets only **_getter_**. Please fix this.

Project requirement | 1
--- | ---
image 

- In this class too, you should have a **_getter_** for `:students`.
```ruby
attr_reader :students
```

File: student.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@ class Student < Person
   def initialize(age, classroom, name = 'Unkown', parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
```
Comment (bobb-Rob) [inline student.rb]: - [x] You are expected to implement a **_belongs-to_** relationship in this class. This `@classroom = classroom` and `classroom.add_student(self)` should be in a method called `classroom`. Example 👇🏾 

```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

And, it should have a getter method using `attr_reader`.
```ruby
attr_reader :classroom
```

Link to the article with examples: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def add_rental(rental)
+    ## has
+    @rentals.push(rental)
+    ## belongs to
+    rental.book = self
+
+    rental.person = self
+  end
```
Comment (bobb-Rob) [inline book.rb]: - [ ] Line 16 means you are assigning **_self_** to `rental.person`, which I think is abnormal. Please go through [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) from the example in a previous lesson. And try to implement this `add_rental` method correctly.

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class Book
+  attr :title, :author
```
Comment (bobb-Rob) [inline book.rb]: - [x] You are expected to have both **_Setters_** and **_getters_** for instance variables (remember about attr_accessor).
This `attr :title, :author` is similar to `attr_reader`, which is only **_getter_**. 

```ruby
attr_accessor :title, :author, :rentals
```

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class Rental
+  attr :date, :book, :person
```
Comment (bobb-Rob) [inline rental.rb]: - [x] You are expected to have both **_Setters_** and **_getters_** for instance variables (remember about attr_accessor).
This `attr :date, :book, :person` is similar to `attr_reader`, which is only **_getter_**. 

```ruby
attr_accessor :date, :book, :person
```

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @yosefHesham , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @yosefHesham,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**